spinner/spinner.go

```diff
@@ -165,6 +166,9 @@ func (s *Spinner) Run() error {
 	if errors.Is(err, tea.ErrProgramKilled) {
 		return nil
 	} else {
+		if actionErr != nil {
```

Comment:
No need for an else above as you return.

Code could be simpler and more readable